runhouse/resources/functions/aws_lambda.py

```diff
@@ -0,0 +1,901 @@
+                    "Resource": "*",
+                    "Effect": "Allow",
+                }
+            ],
+        }
+
+        assume_role_policy_document = {
+            "Version": "2012-10-17",
+            "Statement": [
+                {
+                    "Effect": "Allow",
+                    "Principal": {"Service": "lambda.amazonaws.com"},
+                    "Action": "sts:AssumeRole",
+                }
+            ],
+        }
+
+        try:
+            role_res = self.iam_client.create_role(
+                RoleName=f"{self.name}_Role",
+                AssumeRolePolicyDocument=json.dumps(assume_role_policy_document),
+            )
+            logger.info(f'{role_res["Role"]["RoleName"]} was created successfully.')
+
+        except self.iam_client.exceptions.EntityAlreadyExistsException:
+            role_res = self.iam_client.get_role(RoleName=f"{self.name}_Role")
+
+        self.iam_client.put_role_policy(
+            RoleName=role_res["Role"]["RoleName"],
+            PolicyName=f"{self.name}_Policy",
+            PolicyDocument=json.dumps(role_policy),
+        )
+
+        return role_res
+
+    def _create_new_lambda(self, env_vars):
+        """Creates new AWS Lambda."""
+        logger.info(f"Creating a new Lambda called {self.name}")
+        path = str(Path(self.local_path_to_code[0]).absolute()).split("/")
+        path = ["/" + path_e for path_e in path]
+        path[0] = ""
+        zip_file_name = f"{''.join(path[:-1])}/{self.name}_code_files.zip"
+        zf = zipfile.ZipFile(zip_file_name, mode="w")
+        try:
+            for file_name in self.local_path_to_code:
+                zf.write(file_name, str(Path(file_name).name))
+        except FileNotFoundError:
+            logger.error(f"Could not find {FileNotFoundError.filename}")
+        finally:
+            zf.close()
+        with open(zip_file_name, "rb") as f:
+            zipped_code = f.read()
+
+        # creating a role for the Lambda, using default policy.
+        # TODO [SB]: in the next phase, enable the user to update the default policy
+        role_res = self._create_role_in_aws()
+
+        with self._deploy_lambda_to_aws(
+            role_res, zipped_code, env_vars
+        ) as lambda_config:
+            logger.info(
+                f'{lambda_config["Configuration"]["FunctionName"]} was created successfully.'
+            )
+            Path(zip_file_name).absolute().unlink()
+            return lambda_config
+
+    def to(
+        self,
+        env: Optional[List[str]] = [],
+        cloud: str = "aws_lambda",
+        # Variables below are deprecated
+        reqs: Optional[List[str]] = None,
+        setup_cmds: Optional[List[str]] = [],
+        force_install: Optional[bool] = False,
+    ):
+        """
+        Set up a function on AWS as a Lambda function.
+
+        See the args of the factory method :func:`lambda_function` for more information.
+
+        Example:
```

Comment:
Isn't the more relevant (and suggested) way of using this method something like:

`rh.function(fn=my_fn).to("aws_lambda")`